src/processes/scoring/calc-user-score.js

```diff
@@ -57,11 +57,12 @@ const calcUserScore = async (userDoc) => {
   calcKarmaScore(userDoc._id, user_score);
 
   // checking and sync date_created
-  if (userDoc.created_at === null) {
-    // if created_at is null, set it to created_at in db
+  if (userDoc.created_at === null || userDoc.register_time === null) {
+    // if created_at or register_time is null, set it to created_at in db
     const user = await UsersFunction.getUserByUserId(userDoc._id);
     if (user) {
       userDoc.created_at = user.created_at;
+      userDoc.register_time = user.created_at;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here is not entirely correct. If `userDoc.register_time` is null but `userDoc.created_at` is not, the code will still execute and overwrite `userDoc.created_at` even though it already has a value. This could lead to unnecessary database calls and potential data inconsistency. Consider separating the checks for `created_at` and `register_time`.

```diff
-  if (userDoc.created_at === null || userDoc.register_time === null) {
-    // if created_at or register_time is null, set it to created_at in db
-     const user = await UsersFunction.getUserByUserId(userDoc._id);
-     if (user) {
-       userDoc.created_at = user.created_at;
-       userDoc.register_time = user.created_at;
-     }
-   }

+  const user = await UsersFunction.getUserByUserId(userDoc._id);
+  if (user) {
+    if (userDoc.created_at === null) {
+      userDoc.created_at = user.created_at;
+    }
+    if (userDoc.register_time === null) {
+      userDoc.register_time = user.created_at;
+    }
+  }
```
Also, note that you're setting `userDoc.register_time` to `user.created_at`. If this is intentional and `register_time` should always be the same as `created_at`, then it's fine. But if they should be different, you might want to adjust this logic.


<!-- This is an auto-generated comment by OSS CodeRabbit -->